nrclient/nrclient.go

```diff
@@ -40,26 +45,33 @@ func NewNRClient(cfg config.NRClientConfig, proxyCfg config.ProxyConfig) (*NRCli
 	return nrClient, nil
 }
 
-func (nrClient *NRClient) Send(logRecords []record.LogRecord) (int, error) {
+func (nrClient *NRClient) Send(logRecords []record.LogRecord) (retry bool, err error) {
 	payloads, err := record.PackageRecords(logRecords)
 	if err != nil {
-		return nonRetriableConnectionError, err
+		log.WithField("error", err).Error("Error packaging request")
+		return false, err
 	}
 
 	for _, payload := range payloads {
 		statusCode, err := nrClient.sendPacket(payload)
-		if err != nil || !isSuccesful(statusCode) {
-			return statusCode, err
+		if err != nil {
+			return false, err
 		}
+		if statusCode/100 == 2 {
+			continue
+		}
+		if _, ok := retryableCodesSet[statusCode]; ok {
```

Comment:
I think this can be extracted to a method named `isStatusCodeRetryable` for example, that returns true or false. It will improve the readability on this method and also separate the responsibility of check if we should retry or not for a given status code from the `Send` method.